Compile failed with "invalid conversion"

**Compile pdal failed with error:**
```
FAILED: CMakeFiles/pdalcpp.dir/pdal/XMLSchema.cpp.o 
/usr/bin/c++ -DUNIX -Dpdalcpp_EXPORTS -I/build/pdal/src/PDAL-2.6.0-src/vendor/gtest/include -I/build/pdal/src/PDAL-2.6.0-src/vendor/gtest -I/build/pdal/src/PDAL-2.6.0-src -I/build/pdal/src/PDAL-2.6.0-src/build/include -I/build/pdal/src/PDAL-2.6.0-src/vendor -I/build/pdal/src/PDAL-2.6.0-src/vendor/eigen -I/build/pdal/src/PDAL-2.6.0-src/vendor/nlohmann -I/build/pdal/src/PDAL-2.6.0-src/vendor/utfcpp/source -I/usr/include/libxml2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -g -ffile-prefix-map=/build/pdal/src=/usr/src/debug/pdal -flto=auto -std=c++17 -fPIC -Wno-implicit-fallthrough -Wno-int-in-bool-context -Wno-dangling-else -Wno-noexcept-type -Wall -Wextra -Wpointer-arith -Wcast-align -Wcast-qual -Wno-error=parentheses -Wno-error=cast-qual -Wredundant-decls -Wno-unused-parameter -Wno-unused-variable -Wno-long-long -Wno-unknown-pragmas -Wno-deprecated-declarations -fvisibility=hidden -fvisibility-inlines-hidden -MD -MT CMakeFiles/pdalcpp.dir/pdal/XMLSchema.cpp.o -MF CMakeFiles/pdalcpp.dir/pdal/XMLSchema.cpp.o.d -o CMakeFiles/pdalcpp.dir/pdal/XMLSchema.cpp.o -c /build/pdal/src/PDAL-2.6.0-src/pdal/XMLSchema.cpp
/build/pdal/src/PDAL-2.6.0-src/pdal/XMLSchema.cpp: In member function ‘bool pdal::XMLSchema::validate(xmlDocPtr, const std::string&)’:
/build/pdal/src/PDAL-2.6.0-src/pdal/XMLSchema.cpp:257:9: error: invalid conversion from ‘void (*)(void*, xmlErrorPtr)’ {aka ‘void (*)(void*, _xmlError*)’} to ‘xmlStructuredErrorFunc’ {aka ‘void (*)(void*, const _xmlError*)’} [-fpermissive]
  257 |         &SchemaParserStructuredErrorHandler, m_global_context);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         |
      |         void (*)(void*, xmlErrorPtr) {aka void (*)(void*, _xmlError*)}
In file included from /build/pdal/src/PDAL-2.6.0-src/pdal/XMLSchema.hpp:50,
                 from /build/pdal/src/PDAL-2.6.0-src/pdal/XMLSchema.cpp:34:
```
Tried both on `pdal 2.6.0` and `pdal master branch`, same error.

**Expected behavior**
Build successfully.

**Compile Environment**
```
$ gcc --version
gcc (GCC) 13.2.1 20230801
$ uname -a
Linux  6.6.2-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Mon, 20 Nov 2023 23:17:26 +0000 x86_64 GNU/Linux
$ gdalinfo --version
GDAL 3.7.3, released 2023/10/30

#  compile command
cmake \
      -Bbuild \
      -GNinja \
      -DCMAKE_INSTALL_PREFIX=/usr \
      -DBUILD_PLUGIN_PGPOINTCLOUD=ON \
      -DWITH_COMPLETION=ON
ninja -C build
```

--- a/pdal/XMLSchema.cpp
+++ b/pdal/XMLSchema.cpp
@@ -73,7 +73,11 @@ namespace pdal
 {
 
 void SchemaStructuredErrorHandler
+#if LIBXML_VERSION >= 21200
+(void * /*userData*/, const xmlError *error)
+#else
 (void * /*userData*/, xmlErrorPtr error)
+#endif
 {
     std::ostringstream oss;
 
@@ -98,14 +102,22 @@ void SchemaStructuredErrorHandler
 }
 
 void SchemaParserStructuredErrorHandler
+#if LIBXML_VERSION >= 21200
+(void * /*userData*/, const xmlError *error)
+#else
 (void * /*userData*/, xmlErrorPtr error)
+#endif
 {
     std::cerr << "Schema parsing error: '" << error->message << "' " <<
         "on line " << error->line << std::endl;
 }
 
 void SchemaValidationStructuredErrorHandler
+#if LIBXML_VERSION >= 21200
+(void * /*userData*/, const xmlError *error)
+#else
 (void * /*userData*/, xmlErrorPtr error)
+#endif
 {
     std::cerr << "Schema validation error: '" << error->message << "' " <<
         "on line " << error->line << std::endl;

